pkg/cortex/cortex.go

```diff
@@ -71,10 +72,11 @@ import (
 
 // Config is the root config for Cortex.
 type Config struct {
-	Target      string `yaml:"target"`
-	AuthEnabled bool   `yaml:"auth_enabled"`
-	PrintConfig bool   `yaml:"-"`
-	HTTPPrefix  string `yaml:"http_prefix"`
+	Target      string          `yaml:"target"`
+	Modules     map[string]bool `yaml:"-"`
```

Comment:
I think keeping both `Target string`  and the new `Modules map[string]bool` may lead to bugs in the future. We may forget to check `Modules` instead of `Target`. A better option may be define a CLI flag custom type in `pkg/util/flagext` which is a `[]string` but implements `flag.Value` and `yaml.Marshaller/Unmarshaller` to marshal/unmarshal a comma separated string.

We wouldn't have a map, but checking if a module is within the `[]string` would be as easy as calling `util.StringsContain()`.

What do you think?

Reply:
There's already flagext.StringSlice, should I just use it?

Reply:
`flagext.StringSlice` doesn't support comma-separated argument, only argument with multiple flags. I'm afraid that would be incompatible change in YAML config.